system/BaseSpec.cfc

```diff
@@ -1003,6 +997,13 @@ component {
 	 * @spec  The spec definition
 	 */
 	BaseSpec function runBeforeEachClosures( required suite, required spec ){
+		// re-bind request utilities to the currently executing test before they may be invoked
+		request.testbox.console = 			() => variables.console(argumentCollection=arguments);
+		request.testbox.debug = 			() => variables.debug(argumentCollection=arguments);
+		request.testbox.clearDebugBuffer = 	() => variables.clearDebugBuffer(argumentCollection=arguments);
```

Comment:
Binding the functions as anonymous closures allow `debug` and `clearDebugBuffer` to maintain their reference to `this.$debugBuffer`.

`console`, `print`, and `println` don't need it because they only dump or output, they have no interaction with the spec.  However I also made them anonymous closures for 1) consistency and 2) protection in the case that the methods are refactored.

Re-binding the functions here in runBeforeEachClosures allows them to be associated with the currently executing spec.  This does two things...

1) It fixes a bug where the debug messages are associated only with the first spec to be run (and thus the first bundle to be run).
2) It let's debug calculate the label from the `$this.currentExecutingSpec`.